### PR TITLE
Hide progress on non-ttys

### PR DESCRIPTION
**Pull request by mvz**
Implements #1118.

### CODE REVIEW

**Comment by troessner** [comment]
Do we now have an actual integration test for the --progress flag left? 
I dont think that just specs are enough...

**Comment by mvz** [comment]
> Do we now have an actual integration test for the --progress flag left? 

I'll take a look.

**Comment by mvz** [comment]
Yes, we do: `features/command_line_interface/show_progress.feature`

**Comment by mvz** [comment]
I'll rebase this once #1121 has been merged.

**Comment by troessner** [comment]
>> Yes, we do: features/command_line_interface/show_progress.feature

The problem I have with this is the following: Suppose I'm an innocent reader of our features (or us in 6 months!), just by reading them top down my impression would be that showing the progress is something that is **false by default** and that you have to enable it explicitly.
I mean, in none of the features it has any impact but in one feature it does, and that's the show_progress.feature where we enable it explicitly.
Or am I missing something here?

File: lib/reek/cli/options.rb [hunk 1]
```diff
@@ -57,8 +57,8 @@ def parse
       private
 
       # :reek:UtilityFunction
-      def color_support?
-        $stdout.tty?
+      def piped_output?
```
Comment (troessner) [inline lib/reek/cli/options.rb]: I think the name is confusing. In the cucumber features above nothing gets piped, at least not in the way I - and probably others - understand pipes. Basically the way we run Reek in our features is the default way that involves no pipe. Maybe rename this to "non_tty_output" and put an explanatory comment above?

I really have a shitty understanding of the tty concept which comes from medieval times and I really do not want to read [tty demystified](http://www.linusakesson.net/programming/tty/) again in my life. Like, ever :-/

**Comment by mvz** [comment]
... and it's green!

File: lib/reek/cli/options.rb (hunk 1 above)
Comment (mvz) [inline lib/reek/cli/options.rb]: That makes sense.

**Comment by mvz** [comment]
I see your point, but I'm not sure how we can solve it, since all output of processes running under Aruba will be non-tty. We have a similar problem for `--color`.

**Comment by troessner** [comment]
Maybe add an explanation for this confusing set up in our contributing docs?

**Comment by mvz** [comment]
Ok, I'll add something there, and a comment in the relevant feature files too.

**Comment by mvz** [comment]
I processed all comments, but I'll open another pull request on master for final review and to make Travis do its thing.

**Pull request closed without merging**